### ISSUE TITLE
Two finger zoom on a touch screen behaves erratically in GTK4 branch

### ISSUE THREAD

**Issue by apandada1** [johnfactotum/foliate#1040]
**Describe the bug**
When I try to zoom with two fingers in my touch screen laptop, it usually does not work, and sometimes, it zooms in a little bit, and gets stuck.

Two finger zoom used to work flawlessly in version 2.6.4

**To Reproduce**
1. Open Foliate GTK4 branch on a touch screen computer with multi-touch support.
2. Try to use two-finger zoom while reading an ebook.

There are no error logs in the terminal when I try to zoom in with two fingers.

**Expected behavior**
One should be able to zoom in/out with two fingers.

**Screenshots**

**Version:**
 - Foliate version: Commit https://github.com/johnfactotum/foliate/commit/2e737e604e85ee3d26c3dda87f18885998537073
 - OS/Distribution and version: Arch Linux
 - Desktop environment: KDE Plasma 5.27.5
 - Installation method: AUR package (https://aur.archlinux.org/packages/foliate-git)

**Comment by johnfactotum** [comment]
Yes, touchscreen pinch zoom was intentionally disabled when swiping was added as it interfered with it, and the fact that it still works sometimes is actually a bit concerning. Technically this is a foliate-js bug, which is where the touch events are currently handled. I will try to add it back, though I'm more interested in knowing whether swiping works or not, since I'm not able to test it in Foliate.

**Comment by apandada1** [comment]
> I'm more interested in knowing whether swiping works or not

Yes, swiping (as well as its animation) works perfectly on the gtk4 branch.

---

In version 2.6.4, both swiping and two-finger-zoom worked (although there were no swipe animations in 2.6.4).
It would be great if both these features are simultaneously available on the next version of Foliate.

**Comment by apandada1** [comment]
Both pinch to zoom and swipe are working perfectly. 

Please feel free to ping me if you want me to test any touch-related feature.
